lib/Service/CronService.php

```diff
@@ -402,17 +408,48 @@ private function o(string $line, bool $ln = true): void {
 
 
 	/**
+	 * we assume that calling this method indicate the process was initiated from BackgroundJobs
+	 *
 	 * @return bool
 	 */
-	public function isRealCron(): bool {
-		$mode = $this->configService->getCoreValue('backgroundjobs_mode', '');
+	public function isRunnable(): bool {
+		$mode = strtolower($this->configService->getCoreValue('backgroundjobs_mode', ''));
+		if ($mode !== 'cron' && $mode !== 'webcron') {
+			return false;
+		}
 
 		if (!$this->configService->getAppValueBool(ConfigService::CRON_ENABLED)) {
 			return false;
 		}
 
 		$this->configService->setAppValueBool(ConfigService::CRON_ENABLED, true);
+		$this->ranFromCron = true;
+
+		return ($this->configService->getAppValueInt(ConfigService::CRON_LOCK) < time() - self::LOCK_TIMEOUT);
+	}
+
+
+	/**
+	 * @param bool $verifyTime
+	 *
+	 * @throws JobsTimeSlotException
+	 */
+	public function lockCron(bool $verifyTime = true): void {
```

Comment:
Don't see how this method "locks" anything, or maybe we're not talking mutex style locking.
Looks like it only checks for the execution time and saves it to app data. Am I missing something?

Reply:
I thought this was to prevent parallel manipulation of a restoring point.

Reply:
there is already a lock per restoring point, this one is to lock any other cron process related to the Backup App to be executed (but Event.php which is not an issue)

Reply:
Ok, even though to locking logic seems to be in `isRunnable`.